pandas/core/dtypes/base.py

```diff
@@ -289,6 +289,8 @@ def is_dtype(cls, dtype: object) -> bool:
             return False
         elif isinstance(dtype, cls):
             return True
+        elif hasattr(dtype, "subtype") and isinstance(dtype.subtype, cls):
```

Comment:
@dsaxton I think replacing `dtype.subtype` &#8594;  `getattr(dtype, "subtype", None)` like you said does silence that `mypy` complaint